server/app/services/program/ProgramServiceImpl.java

```diff
@@ -210,7 +216,17 @@ public ErrorAnd<ProgramDefinition, CiviFormError> updateLocalization(
   }
 
   private boolean hasProgramNameCollision(String programName) {
-    return getActiveAndDraftPrograms().getProgramNames().contains(programName);
+    ImmutableSet<String> programNames = getActiveAndDraftPrograms().getProgramNames();
```

Comment:
While you're not changing this part; how about including obsolete ones (EG all programs) for good measure?  It's technically a valid state and we could allow for archiving programs in the future.

Reply:
done

Reply:
you might notice that there's a slug column, and you might wonder "why not query that column instead of the name column", and my answer is: because the slug column is not guaranteed to be populated like the name column is and changing that is more than i'd like to do in this PR.

it brings me no joy to mention this, as it brought me no joy to discover it.

Reply:
Seems like a reasonable thing to ask Seattle to shore up, it could be done with a little manual work too I would hope.  Shall we ask them about it?

Reply:
Sounds good to me!